src/Bundle.ts

```diff
@@ -178,18 +178,20 @@ export default class Bundle {
 			this.outputOptions,
 			inputBase
 		);
-		const chunks: Chunk[] = [];
+		const executableModule = inlineDynamicImports
+		? [{ alias: null, modules: includedModules }]
+		: preserveModules
+			? includedModules.map(module => ({ alias: null, modules: [module] }))
+			: getChunkAssignments(
+					this.graph.entryModules,
+					manualChunkAliasByEntry,
+					experimentalMinChunkSize,
+					this.inputOptions.onLog
+				)
+		const chunks: Chunk[] = new Array(executableModule.length);
 		const chunkByModule = new Map<Module, Chunk>();
-		for (const { alias, modules } of inlineDynamicImports
-			? [{ alias: null, modules: includedModules }]
-			: preserveModules
-				? includedModules.map(module => ({ alias: null, modules: [module] }))
-				: getChunkAssignments(
-						this.graph.entryModules,
-						manualChunkAliasByEntry,
-						experimentalMinChunkSize,
-						this.inputOptions.onLog
-					)) {
+		let index = 0
+		for (const { alias, modules } of executableModule) {
 			sortByExecutionOrder(modules);
```

Comment:
I would not expect too much improvement from this change as this is not a hot code path, it is just run once for the entire build. If you want to see results, I would rather focus on changes in the AST nodes.